pkg/crypto/cryptoutil/kek_labelers.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptoutil
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"strings"
+
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+// ComponentPrefixKEKLabeler is a ComponentKEKLabeler that joins the component prefix, separator and host.
+type ComponentPrefixKEKLabeler struct {
+	Separator string
+}
+
+func hostFromAddr(addr string) (host string) {
+	if url, err := url.Parse(addr); err == nil && url.Host != "" {
+		host = url.Host
+	} else {
+		host = addr
+	}
+	if h, _, err := net.SplitHostPort(host); err == nil {
+		host = h
+	}
+	return
+}
+
+// NsKEKLabel returns a KEK label in the form `ns:netID:host` from the given NetID and address. Empty parts are omitted.
+func (c ComponentPrefixKEKLabeler) NsKEKLabel(ctx context.Context, netID *types.NetID, addr string) string {
```

Comment:
Shouldn't KEK labels include a sequence number to allow for rotation? Same for AS below.

Reply:
Any `ComponentKEKLabeler` that the key vault provides can do that. The static/in-memory key vault simply uses a prefix, but I can imagine a more dynamic key vault using all kinds of rotation schemes. There’s no point in a sequence number when the key vault is static